documentation/Templates.md

```diff
@@ -67,6 +67,8 @@ Additionally to the [tags and filters](https://stencil.fuller.li/en/latest/built
 | Node | Description |
 | ---- | ----------- |
 | `{% set my_capture_variable %}`<br>`{% endset %}` | node pair allowing you to capture the rendering of nodes inside it, so you can reuse it later by reusing `{{ my_capture_variable }}`.<br>This is useful if you intend to reuse the same content at multiple places — like the `sceneID` is for the `storyboards-default.stencil` template for example. |
+| `{% func my_block a b c %}`<br>`{{args.a}} - {{args.b}} - {{args.c}}`<br>`{% endfunc %}` | node pair allowing you to define a reusable block of nodes inside (like a function), so you can reuse it later.<br>In contrast to the `set` tag, the capturing for `func` is lazy, it will only be processed when actually called. When processed, the block will provide a variable `args` containing all the parameter values.<br>Similar to the `for` tag, this tag provides it's own context scope when called, so variables inside of it are not available outside of it. |
```

Comment:
I think I remember back in the day when I first implemented `SetNode` that I did that in order to avoid having secondary files and have stand-alone templates instead, especially for reusable parts that are short.

I think the same might apply here for func/call in the sense that sure `include` makes sense and can be really nice to allow people to override only parts of the templates, but:

1. sometimes the parts we reuse are so small that it feels not worth it to separate them in dedicated files just for a single node
2. but more importantly SwiftGen templates offer _alternatives_, unlike templates to generate a web site like with Jekyll for example. What I mean by that is that the files we have in `/usr/local/share/swiftgen/templates` are more "1 file = one template alternative to choose from, by using the `-t` flag during invocation", unlike a HTML website where the set of template files are more various parts to construct a single web site and you'll always use the same templates every time you regenerate your website.

Not sure that makes sense?

Anyway, that could still be interesting to see if using`{% include %}` in some of our cases is worth it (for the car you point out maybe it is indeed) but people should then be careful to use template parts that goes together when they select one over the other with `-t`. And if we do that we might end up with template names like `strings-dot-syntax-swift3[case].inc.stencil` or similar to disambiguate… would be hard not to be lost in all the template alternatives otherwise…